Update ERC-7943: Align with ERC-3643 naming conventions and remove ERC7943NotAllowedTransfer

ernestognw opened the pull request:
Aligned ERC-7943 with ERC-3643 naming conventions for better RWA ecosystem compatibility:

- Renamed `isTransferAllowed` → `canTransfer` (matches ERC-3643, ERC-7518, ERC-7751)
- Renamed `forceTransfer` → `forcedTransfer` (matches ERC-3643)
- Renamed `getFrozen` → `getFrozenTokens` (matches ERC-3643)
- Renamed `setFrozen` → `setFrozenTokens` (for consistency)
- Removed generic `ERC7943NotAllowedTransfer` error to encourage specific error handling

Updated all specifications, examples, and documentation. No breaking changes to core functionality, only improved naming consistency across RWA ERCs.

xaler5 commented on ERCS/erc-7943.md:
```diff
@@ -150,34 +145,34 @@ It provides a way to freeze or unfreeze assets held by a specific user. This is
 The contract MUST implement the [ERC-165](./eip-165.md) `supportsInterface` function and MUST return true for the `bytes4` value `0xf35fc3be` being it the `interfaceId` of the [ERC-7943](./eip-7943.md).
 
-Given the agnostic nature of the standard on the specific base token standard being used, the implementation SHOULD use `tokenId = 0` for [ERC-20](./eip-20.md) based implementations, and `amount = 1` for [ERC-721](./eip-721.md) based implementations on `ForcedTransfer` and `Frozen` events, `ERC7943NotAllowedTransfer` and `ERC7943InsufficientUnfrozenBalance` errors, and `forceTransfer`, `setFrozen ...
+Given the agnostic nature of the standard on the specific base token standard being used, the implementation SHOULD use `tokenId = 0` for [ERC-20](./eip-20.md) based implementations, and `amount = 1` for [ERC-721](./eip-721.md) based implementations on `ForcedTransfer` and `Frozen` events, `ERC7943InsufficientUnfrozenBalance` errors, and `forcedTransfer`, `setFrozen`, `getFrozenTokens` and `canTr ...
```
```suggestion
Given the agnostic nature of the standard on the specific base token standard being used, the implementation SHOULD use `tokenId = 0` for [ERC-20](./eip-20.md) based implementations, and `amount = 1` for [ERC-721](./eip-721.md) based implementations on `ForcedTransfer` and `Frozen` events, `ERC7943InsufficientUnfrozenBalance` error, and `forcedTransfer`, `setFrozen`, `getFrozenTokens` and `canTransfer` functions. Integrators MAY decide to not enforce this, however the standard discourages it. This is considered a minor tradeoff for having a unique standard interface for different token standards without overlapping syntaxes.
```

xaler5 commented on ERCS/erc-7943.md:
```diff
@@ -98,7 +93,7 @@ interface IERC7943 /*is IERC165*/ {
     /// @param tokenId The ID of the token.
     /// @return amount The amount of `tokenId` tokens currently frozen for `user`.
-    function getFrozen(address user, uint256 tokenId) external view returns (uint256 amount);
+    function getFrozenTokens(address user, uint256 tokenId) external view returns (uint256 amount);
```
At this point if these functions were `set/get Frozen` I believe it makes sense to rename `setFrozen` to `setFrozenTokens` too. Even if I believe that "tokens" is not the right word to use since ERC-721/1155 go by single units, for the sake of naming consistency I can accept that.

xaler5 commented on ERCS/erc-7943.md:
```diff
@@ -150,34 +145,34 @@ It provides a way to freeze or unfreeze assets held by a specific user. This is
 
-- **`forceTransfer`**: This term was selected for its neutrality. While names like "confiscation," "revocation," or "recovery" describe specific motivations, `forceTransfer` purely denotes the direct action of transferring assets, irrespective of the underlying reason.
-- **`setFrozen` / `getFrozen`**: These names were chosen for managing transfer restrictions.
+- **`forcedTransfer`**: This term was selected for its neutrality. While names like "confiscation," "revocation," or "recovery" describe specific motivations, `forcedTransfer` purely denotes the direct action of transferring assets, irrespective of the underlying reason.
```
I'd add a note on forcedTransfer/canTransfer to leave trace of the fact that these name where chosen because of consistency with all the other three mentioned ERCs. I believe that the "Notes on naming" section is important to establish a bit of clarity in the RWA space.

tinom9 commented:
As I mentioned in my EM post, I think differences between ERC3643's `canTransfer` and ERC7943's `isTransferAllowed` should be documented.

ernestognw commented:
> Thanks for this @ernestognw ! I really appreciate the efforts :) I've left some comments to address. Additionally if you're up for that, fell free to reflect these changes in the reference implementation https://github.com/xaler5/uRWA
> 
> Otherwise I'll do it in a couple of days anyway.
> 
> Thanks again !

No worries. Just updated addressing your comments and also published a PR to update your reference implementation:
https://github.com/xaler5/uRWA/pull/4
